From dc22fd13bff1c38888ac16bc8f8fea8117ef27bd Mon Sep 17 00:00:00 2001
From: Jan Friesse <jfriesse@redhat.com>
Date: Fri, 20 Jul 2012 09:26:14 +0000
Subject: [PATCH] cpg: Be more verbose for procjoin message

Signed-off-by: Jan Friesse <jfriesse@redhat.com>
Reviewed-by: Steven Dake <sdake@redhat.com>
---
 services/cpg.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/services/cpg.c b/services/cpg.c
index d7a26f2..aac7a90 100644
--- a/services/cpg.c
+++ b/services/cpg.c
@@ -1088,7 +1088,10 @@ static void message_handler_req_exec_cpg_procjoin (
 {
 	const struct req_exec_cpg_procjoin *req_exec_cpg_procjoin = message;
 
-	log_printf(LOGSYS_LEVEL_DEBUG, "got procjoin message from cluster node %d\n", nodeid);
+	log_printf(LOGSYS_LEVEL_DEBUG, "got procjoin message from cluster node %d (%s) for pid %u",
+		nodeid,
+		api->totem_ifaces_print(nodeid),
+		(unsigned int)req_exec_cpg_procjoin->pid);
 
 	do_proc_join (&req_exec_cpg_procjoin->group_name,
 		req_exec_cpg_procjoin->pid, nodeid,
-- 
1.7.1

