From bc130cb10c4ec014829286ba61c0de544f82656d Mon Sep 17 00:00:00 2001
From: Jan Friesse <jfriesse@redhat.com>
Date: Tue, 16 Apr 2013 16:09:29 +0200
Subject: [PATCH] Install sysconfig/corosync-notifyd in specfile

Signed-off-by: Jan Friesse <jfriesse@redhat.com>
Reviewed-by: Fabio M. Di Nitto <fdinitto@redhat.com>
---
 corosync.spec.in |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/corosync.spec.in b/corosync.spec.in
index 04970de..980982b 100644
--- a/corosync.spec.in
+++ b/corosync.spec.in
@@ -100,6 +100,10 @@ install -m 644 %{_builddir}/%{name}-%{version}/conf/corosync-signals.conf %{buil
 rm -f %{buildroot}%{_libdir}/*.a
 # drop docs and html docs for now
 rm -rf %{buildroot}%{_docdir}/*
+# /etc/sysconfig/corosync-notifyd
+mkdir -p %{buildroot}%{_sysconfdir}/sysconfig
+install -m 644 tools/corosync-notifyd.sysconfig.example \
+   %{buildroot}%{_sysconfdir}/sysconfig/corosync-notifyd
 
 %clean
 rm -rf %{buildroot}
@@ -137,6 +141,7 @@ fi
 %dir %{_sysconfdir}/corosync/uidgid.d
 %config(noreplace) %{_sysconfdir}/corosync/corosync.conf.example
 %config(noreplace) %{_sysconfdir}/corosync/corosync.conf.example.udpu
+%config(noreplace) %{_sysconfdir}/sysconfig/corosync-notifyd
 %if %{with dbus}
 %{_sysconfdir}/dbus-1/system.d/corosync-signals.conf
 %endif
-- 
1.7.1

