From e69f322d52a967e19ac6cf4eba60c54fc64b1688 Mon Sep 17 00:00:00 2001
From: Jan Friesse <jfriesse@redhat.com>
Date: Fri, 30 Nov 2012 12:28:27 +0100
Subject: [PATCH] log: Avoid deadlock caused by previous commit

Signed-off-by: Jan Friesse <jfriesse@redhat.com>
---
 exec/logsys.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/exec/logsys.c b/exec/logsys.c
index b82c647..c4611bc 100644
--- a/exec/logsys.c
+++ b/exec/logsys.c
@@ -443,13 +443,13 @@ static void log_printf_to_logs (
 	subsysid = LOGSYS_DECODE_SUBSYSID(rec_ident);
 	level = LOGSYS_DECODE_LEVEL(rec_ident);
 
-	pthread_mutex_lock (&logsys_config_mutex);
-
 	if (!((LOGSYS_DECODE_RECID(rec_ident) == LOGSYS_RECID_LOG) ||
 	      (logsys_loggers[subsysid].trace1_allowed && LOGSYS_DECODE_RECID(rec_ident) == LOGSYS_RECID_TRACE1))) {
 		return;
 	}
 
+	pthread_mutex_lock (&logsys_config_mutex);
+
 	while ((c = format_buffer[format_buffer_idx])) {
 		cutoff = 0;
 		if (c != '%') {
-- 
1.7.1

