From 3e3231375c29d562af050a48f18b301bf1cdca82 Mon Sep 17 00:00:00 2001
From: Jan Friesse <jfriesse@redhat.com>
Date: Wed, 6 Mar 2013 13:50:49 +0100
Subject: [PATCH] totempg: Make iov_delv local variable

Signed-off-by: Jan Friesse <jfriesse@redhat.com>
Reviewed-by: Fabio M. Di Nitto <fdinitto@redhat.com>
---
 exec/totempg.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/exec/totempg.c b/exec/totempg.c
index d41e548..0649cab 100644
--- a/exec/totempg.c
+++ b/exec/totempg.c
@@ -226,8 +226,6 @@ static int fragment_size = 0;
 
 static int fragment_continuation = 0;
 
-static struct iovec iov_delv;
-
 static int totempg_waiting_transack = 0;
 
 static unsigned int totempg_max_handle = 0;
@@ -628,6 +626,7 @@ static void totempg_deliver_fn (
 	int start;
 	const char *data;
 	int datasize;
+	struct iovec iov_delv;
 
 	assembly = assembly_ref (nodeid);
 	assert (assembly);
-- 
1.7.1

